dataline-server/src/main/java/io/dataline/server/apis/ConfigurationApi.java

```diff
@@ -116,177 +121,194 @@ public ConfigurationApi(final Path dbRoot, BasicDataSource connectionPool) {
-        destinationImplementationIdRequestBody);
+    return execute(() -> schedulerHandler.checkDestinationImplementationConnection(destinationImplementationIdRequestBody));
   }
 
   // CONNECTION
 
   @Override
   public ConnectionRead createConnection(@Valid ConnectionCreate connectionCreate) {
-    return connectionsHandler.createConnection(connectionCreate);
+    return execute(() -> connectionsHandler.createConnection(connectionCreate));
   }
 
   @Override
   public ConnectionRead updateConnection(@Valid ConnectionUpdate connectionUpdate) {
-    return connectionsHandler.updateConnection(connectionUpdate);
+    return execute(() -> connectionsHandler.updateConnection(connectionUpdate));
   }
 
   @Override
   public ConnectionReadList listConnectionsForWorkspace(@Valid WorkspaceIdRequestBody workspaceIdRequestBody) {
-    return connectionsHandler.listConnectionsForWorkspace(workspaceIdRequestBody);
+    return execute(() -> connectionsHandler.listConnectionsForWorkspace(workspaceIdRequestBody));
   }
 
   @Override
   public ConnectionRead getConnection(@Valid ConnectionIdRequestBody connectionIdRequestBody) {
-    return connectionsHandler.getConnection(connectionIdRequestBody);
+    return execute(() -> connectionsHandler.getConnection(connectionIdRequestBody));
   }
 
   @Override
   public ConnectionSyncRead syncConnection(@Valid ConnectionIdRequestBody connectionIdRequestBody) {
-    return schedulerHandler.syncConnection(connectionIdRequestBody);
+    return execute(() -> schedulerHandler.syncConnection(connectionIdRequestBody));
   }
 
   // JOB HISTORY
 
   @Override
   public JobReadList listJobsFor(@Valid JobListRequestBody jobListRequestBody) {
-    return jobHistoryHandler.listJobsFor(jobListRequestBody);
+    return execute(() -> jobHistoryHandler.listJobsFor(jobListRequestBody));
   }
 
   @Override
   public JobInfoRead getJobInfo(@Valid JobIdRequestBody jobIdRequestBody) {
-    return jobHistoryHandler.getJobInfo(jobIdRequestBody);
+    return execute(() -> jobHistoryHandler.getJobInfo(jobIdRequestBody));
   }
 
   // WEB BACKEND
 
   @Override
   public WbConnectionReadList webBackendListConnectionsForWorkspace(@Valid WorkspaceIdRequestBody workspaceIdRequestBody) {
-    return webBackendConnectionsHandler.webBackendListConnectionsForWorkspace(workspaceIdRequestBody);
+    return execute(() -> webBackendConnectionsHandler.webBackendListConnectionsForWorkspace(workspaceIdRequestBody));
   }
 
   @Override
   public WbConnectionRead webBackendGetConnection(@Valid ConnectionIdRequestBody connectionIdRequestBody) {
-    return webBackendConnectionsHandler.webBackendGetConnection(connectionIdRequestBody);
+    return execute(() -> webBackendConnectionsHandler.webBackendGetConnection(connectionIdRequestBody));
+  }
+
+  private <T> T execute(HandlerCall<T> call) {
+    try {
+      return call.call();
+    } catch (ConfigNotFoundException e) {
+      throw new KnownException(
+          HttpStatus.NOT_FOUND_404,
```

Comment:
This is a behavior change. Before we were sending 422. Let me know if 404 is more correct.

Reply:
422 is safer for now. remember that the `getXConfig` can be called internally in any of these handlers. e.g. connections/create might internally retrieve the `sourceImplementation` that it's going to be associated with to validate it exists. If it fails to find that `sourceImplementation` I believe the correct https status code would be 422. There are cases though where we try to fetch a config and failure to do so should return a 404. In our current scheme we sometimes return 422 when we should probably be returning a 404. To get this right we need to go to each invocation for `configPersistence.getConfig(...)` and throw the appropriate exception either 404 or 422.

Reply:
I will restore 422 but I think it is wrong. If what you describe happens then the code that validates should catch the `ConfigNotFoundException` and rethrow it as a ValidationException.

Reply:
created https://github.com/datalineio/dataline/issues/191